src/main/java/az/ingress/ms/user/controller/InternalUserController.java

```diff
@@ -0,0 +1,63 @@
+package az.ingress.ms.user.controller;
+
+import az.ingress.ms.user.model.request.SignInRequest;
+import az.ingress.ms.user.model.request.SignUpRequest;
+import az.ingress.ms.user.model.request.UserProfileUpdateRequest;
+import az.ingress.ms.user.model.response.SignInResponse;
+import az.ingress.ms.user.model.response.UserResponse;
+import az.ingress.ms.user.service.abstraction.UserService;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+import static az.ingress.ms.user.model.constants.HeaderConstants.USER_ID;
+import static lombok.AccessLevel.PRIVATE;
+import static org.springframework.http.HttpStatus.CREATED;
+import static org.springframework.http.HttpStatus.NO_CONTENT;
+
+@RestController
+@RequestMapping("internal/v1/users")
+@RequiredArgsConstructor
+@FieldDefaults(level = PRIVATE, makeFinal = true)
+public class InternalUserController {
+
+    UserService userService;
+
+    @PostMapping("sign-up")
+    @ResponseStatus(CREATED)
+    public void createUser(@RequestBody SignUpRequest signUpRequest) {
+        userService.createUser(signUpRequest);
+    }
+
+    @PostMapping("sign-in")
+    public SignInResponse login(@RequestBody SignInRequest signInRequest) {
+        return userService.login(signInRequest);
+    }
+
+    @PatchMapping("/profile")
+    @ResponseStatus(NO_CONTENT)
+    public void updateProfile(@RequestHeader(USER_ID) Long userId, @RequestBody UserProfileUpdateRequest userProfileUpdateRequest) {
```

Comment:
not exist MultipartFile for accepting photo. You should accept photo via multipart and convert it to base64 and save to db